sinks/signalfx/signalfx.go

```diff
@@ -250,7 +261,12 @@ func (sfx *SignalFxSink) clientByTagUpdater() {
 
 	ticker := time.NewTicker(sfx.dynamicKeyRefreshPeriod)
 	for range ticker.C {
-		tokens, err := fetchAPIKeys(sfx.httpClient, sfx.apiEndpoint, sfx.defaultToken)
+		sessionToken, err := getSessionToken(sfx.httpClient, sfx.apiEndpoint, sfx.email, sfx.password)
```

Comment:
Nit: Should we do any empty username/password validation before or in this method if they enabled dynamic tokens but forgot to supply config. May be not needed if error looks descriptive enough for empty username/password.

Reply:
The error is a generic invalid credentials error. Sounds reasonable but will add the check to the Sink constructor instead of here.